### PR TITLE
Add surface force postprocessor

### LINKED ISSUE #24909: Add postprocessors to measure lift and drag coefficients on an immersed object.
## Reason
We don't have a streamlined way for measuring lift and drag coefficients on moving objects.

## Design
Add two postprocessors.

## Impact
Will enable flow control over a cylinder.

### PR DESCRIPTION
**Pull request by grmnptr**
Closes #24909

### CODE REVIEW

File: modules/navier_stokes/include/postprocessors/IntegralDirectedSurfaceForce.h [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+// MOOSE includes
+#include "SideIntegralPostprocessor.h"
+#include "MathFVUtils.h"
+
+/**
+ * This postprocessor which computes the directed force coming from friction and pressure
```
Comment (lindsayad) [inline modules/navier_stokes/include/postprocessors/IntegralDirectedSurfaceForce.h:17]:
```suggestion
 * Postprocessor which computes the directed force coming from friction and pressure
```

File: modules/navier_stokes/include/postprocessors/IntegralDirectedSurfaceForce.h [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  /// Velocity components
+  std::vector<const Moose::Functor<Real> * const> _vel_components;
+
+  /// The dynamic viscosity
+  const Moose::Functor<Real> & _mu;
+
+  /// Pressure field
+  const Moose::Functor<Real> & _pressure;
+
+  /// The direction in which the force is measured is measured
```
Comment (lindsayad) [inline modules/navier_stokes/include/postprocessors/IntegralDirectedSurfaceForce.h:45]:
```suggestion
  /// The direction in which the force is measured
```

File: modules/navier_stokes/src/postprocessors/IntegralDirectedSurfaceForce.C [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+registerMooseObject("NavierStokesApp", IntegralDirectedSurfaceForce);
+
+InputParameters
+IntegralDirectedSurfaceForce::validParams()
+{
+  InputParameters params = SideIntegralPostprocessor::validParams();
+  params.addClassDescription(
+      "Computes the directed force coming from friction and pressure differences on a surface. One "
+      "can "
+      "use this object for the computation of the drag and lift coefficient as well.");
```
Comment (lindsayad) [inline modules/navier_stokes/src/postprocessors/IntegralDirectedSurfaceForce.C:27]: spacing funny

File: modules/navier_stokes/src/postprocessors/IntegralDirectedSurfaceForce.C [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+  const auto state = determineState();
+  const auto face_arg =
+      Moose::FaceArg({fi, Moose::FV::LimiterType::CentralDifference, true, false, nullptr});
+  const auto elem_arg = Moose::ElemArg({fi->elemPtr(), false});
+
+  RealTensorValue pressure_term;
+  RealVectorValue cell_velocity = 0;
+  RealVectorValue face_velocity = 0;
+  Real pressure = _pressure(face_arg, state);
+  Real mu = _mu(face_arg, state);
```
Comment (lindsayad) [inline modules/navier_stokes/src/postprocessors/IntegralDirectedSurfaceForce.C:77]:
```suggestion
  const Real pressure = _pressure(face_arg, state);
  const Real mu = _mu(face_arg, state);
```

File: modules/navier_stokes/src/postprocessors/IntegralDirectedSurfaceForce.C [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+  for (auto i : make_range(_mesh.dimension()))
```
Comment (lindsayad) [inline modules/navier_stokes/src/postprocessors/IntegralDirectedSurfaceForce.C:78]:
```suggestion
  for (const auto i : make_range(_mesh.dimension()))
```

File: modules/navier_stokes/src/postprocessors/IntegralDirectedSurfaceForce.C [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+  {
+    cell_velocity(i) = (*_vel_components[i])(elem_arg, state);
+    face_velocity(i) = (*_vel_components[i])(face_arg, state);
+    pressure_term(i, i) = -pressure;
+  }
+
+  auto sheer_force = -mu *
```
Comment (lindsayad) [inline modules/navier_stokes/src/postprocessors/IntegralDirectedSurfaceForce.C:85]:
```suggestion
  const auto sheer_force = -mu *
```

File: modules/navier_stokes/src/postprocessors/IntegralDirectedSurfaceForce.C [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+                      (cell_velocity - face_velocity) * fi->normal() * fi->normal()) /
+                     std::abs(fi->dCN() * fi->normal());
+
+  return (sheer_force * _direction + pressure_term * fi->normal() * _direction);
+}
+
+Real
+IntegralDirectedSurfaceForce::computeQpIntegral()
+{
+  return 0.0;
```
Comment (lindsayad) [inline modules/navier_stokes/src/postprocessors/IntegralDirectedSurfaceForce.C:96]: should we error if we get here?

File: modules/navier_stokes/doc/content/source/postprocessors/IntegralDirectedSurfaceForce.md [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+
+!equation
+F_d = \int\limits_S (\sigma \vec{n}) \cdot \vec{d} ~dS
+
+where $\sigma$ is the Cauchy stress tensor. One can use this in combination with other postprocessors
+to compute representative drag and lift coefficients.
+
+## Example input syntax
+
+In this case, the lift and drag coefficients are computed on the surface of a cylinder.
```
Comment (lindsayad) [inline modules/navier_stokes/doc/content/source/postprocessors/IntegralDirectedSurfaceForce.md:18]: Can you explain a little about the sign and vector choices of the `principal_direction` in the example? E.g. why do we choose the -x direction for calculation of the drag force? And -y direction for lift? This is probably obvious to some people, but it's not to me

**Review by lindsayad — CHANGES_REQUESTED** [review]

File: modules/navier_stokes/doc/content/source/postprocessors/IntegralDirectedSurfaceForce.md (hunk 8 above)
Comment (grmnptr) [reply modules/navier_stokes/doc/content/source/postprocessors/IntegralDirectedSurfaceForce.md:18]: Ahhh I almost forgot this, I wanted to follow the description in this:
https://www.ljll.math.upmc.fr/hecht/ftp/NSF03/old/2014-15/w8/SchaeferTurek1996.pdf

But I added a negative sign wrongly due to a false perception of the cauchy stress tensor acting on the fluid. But we are actually acting on the solid (counter-force).

**Review by lindsayad — APPROVED** [review]

**Pull request merged**